fundingmanager.go

```diff
@@ -1386,6 +1444,13 @@ func (f *fundingManager) handleFundingSigned(fmsg *fundingSignedMsg) {
 		return
 	}
 
+	// At this point we have received the last funding message from the
+	// remote peer before we broadcast the funding transaction so set
+	// the timestamp to the zero value. This indicates to the zombie
+	// sweeper that this reservation should not be pruned, as it is now
+	// waiting for the funding transaction
+	resCtx.lastUpdated = time.Time{}
```

Comment:
set it to a high future value instead?

Reply:
Actually, I think all the timestamp updates can be moved to the end of the corresponding methods (just use a `defer` just after getting the context?). This to ensure consistency, and because the timeout is long enough for it to not make any difference.

Reply:
I like the idea of deferring the call. I will implement it that way and make it consistent across each step in the funding flow. Regarding setting it to a high future value, I think it is cleaner and more maintainable to set it to the time zero value. The time zero value is `January 1, year 1, 00:00:00 UTC` a time that will never be set by `time.Now`. If it is set to a future time, eventually this time would need to be changed. It also simplifies creating the time object (instantiate an empty `time.Time` struct, rather than having to specify the time down to the nanosecond via the `time.Date` function) and comparing the time object (just call `IsZero()`).

Reply:
Yeah, that's reasonable. I was thinking of something similar to Swift's `Date.distantFuture`, but doesn't look like Go has the equivalent :p